### PR TITLE
CHK-90: Add support for disposable addresses and snd purchase scenarios

### PR DESCRIPTION
**Pull request by lucasecdb**
#### What problem is this solving?

This PR updates the shipping flow to accomodate the second purchase scenarios (adding a new address, edit the receiver name when not authorized, etc). This also implements the support for disposable addresses which is required for the second purchase flow to function correctly.

Depends on vtex/checkout-container#7

#### How should this be manually tested?

[Workspace](https://sndpurchase--checkoutio.myvtex.com/).

There are a few tests I recommend to see if everything is fine:

1. Editing a receiver name when identified (not authorized).

   1. [Add a item to cart](https://sndpurchase--checkoutio.myvtex.com/cart/add?sku=289).
   2. Proceed to checkout.
   3. Input a second purchase email when prompted.
   4. Edit the shipping info.
   5. Press the icon to edit the receiver name on the SLA selection step.
   6. The VTEX ID modal should appear.

2. Adding a disposable address and being required for edit the receiver name.

   You should follow the same 4 steps as the test above.

   1. When editing the shipping info, edit the shipping address.
   2. Select to create a new address from the address list.
   3. Use any zipcode to create a new disposable address.
   4. Select any SLA.
   5. Fill the number of the address.
   6. You should end up in the step to edit the address receiver name, and should not see the masked name in the input.

3. Editing a disposable address as a first purchase user.

   This test is a bit tricky and _really_ edge case, but it's worth testing NTL.
   Follow until the step 3 of the test nº 2.

   1. Before editing the address, logout.
   2. Enter a first purchase email in the identification page.
   3. Proceed to edit the shipping info.
   4. Select any SLA.
   5. Input a number in the number field. You should be able to see the checkbox for the receiver name here.
   6. If the checkbox is checked, you should be redirected to payment, else you should be in the receiver info edit step.

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [x] Linked this PR to a Jira story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

**Review by marcoskwkm — APPROVED** [review]
noice

**Comment by lucasecdb** [comment]
rebased + updated typings packages

**Review by jeffersontuc — APPROVED** [review]
LGTM

File: react/shippingStateMachine/shippingStateMachine.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import { assign, Machine } from 'xstate'
 
-import {
+import type {
```
Comment (marcoskwkm) [inline react/shippingStateMachine/shippingStateMachine.ts:3]: nice

**Review by marcoskwkm — APPROVED** [review]
